src/command/dev/mod.rs

```diff
@@ -63,38 +74,140 @@ pub struct SupergraphOpts {
     /// If you start multiple `rover dev` processes on the same address and port, they will communicate with each other.
     ///
     /// If you start multiple `rover dev` processes with different addresses and ports, they will not communicate with each other.
-    #[arg(long, short = 'p', default_value = "3000")]
-    supergraph_port: u16,
+    #[arg(long, short = 'p')]
+    supergraph_port: Option<u16>,
 
     /// The address the graph router should listen on.
     ///
     /// If you start multiple `rover dev` processes on the same address and port, they will communicate with each other.
     ///
     /// If you start multiple `rover dev` processes with different addresses and ports, they will not communicate with each other.
-    #[arg(long, default_value = "127.0.0.1")]
-    supergraph_address: String,
+    #[arg(long)]
+    supergraph_address: Option<String>,
+
+    /// The path to a router configuration file. If the file is empty, a default configuration will be written to that file.
+    ///
+    /// For information on the format of this file, please see https://www.apollographql.com/docs/router/configuration/overview/#yaml-config-file.
+    #[arg(long = "router-config", conflicts_with_all = ["supergraph_port", "supergraph_address"])]
```

Comment:
I didn't know about `conflicts_with_all`, that's super cool

Reply:
it is cool! useful for mutually exclusive args.